reproduce-figures/prepare-scpca-portal-data.R

```diff
@@ -0,0 +1,309 @@
+  )
+)
+opts <- parse_args(OptionParser(option_list = option_list))
+
+
+# Genes whose expression should be kept in consensus cell type marker gene TSVs
+marker_gene_ref_file <- "https://raw.githubusercontent.com/AlexsLemonade/OpenScPCA-analysis/refs/tags/v0.2.2/analyses/cell-type-consensus/references/validation-markers.tsv"
+
+# Setup --------------------
+
+# Check files and directories
+stopifnot(
+  "Portal projects directory not found" = dir.exists(opts$portal_projects_dir), 
+  "Merged SCE for project SCPCP000003 not found" = file.exists(opts$merged_sce_path),
+  "Output directory not specified" = !is.null(opts$output_dir), 
+  "Project whitelist could not be found" = file.exists(opts$project_whitelist)
+)
+
+# Define and check output directories
+output_dir_s3_files <- file.path(opts$output_dir, "s3_files")
+output_dir_analysis_data <- file.path(opts$output_dir, "scpca_data")
+
+if ((dir.exists(output_dir_s3_files) | dir.exists(output_dir_analysis_data)) & !opts$overwrite) {
+  stop("Output directories already exist. To overwrite them, use the --overwrite flag.")
+}
+
+# Define additional nested directories and relevant sample/library ids
+output_dir_consensus_files <- file.path(output_dir_s3_files, "cell-type-consensus-results")
+output_dir_celltype_results <- file.path(output_dir_s3_files, "celltype_results")
+
+# Create directories which will be assumed to exist later
+fs::dir_create(output_dir_analysis_data)
+fs::dir_create(output_dir_celltype_results)
+
+# Define marker genes for recreating expression matrices used for consensus cell type dot plots
+marker_genes <- readr::read_tsv(marker_gene_ref_file) |>
+  dplyr::pull(ensembl_gene_id) |>
+  unique()
+
+# These are all output directories at the top-level of s3_files
+### Some need just processed, and some also need (un)filtered, but scripts specify which one
+### We can therefore retain all RDS files when copying; extra files will not be used either way
+standalone_sample_dirs <- c(
+  "SCPCS000001", 
+  "SCPCS000216", 
+  "SCPCS000251", 
+  "SCPCS000264"
+)
+
+# The samples' processed.rds files should be saved to output_dir_celltype_results
+celltype_results_samples <- c(
+  "SCPCS000001",
+  "SCPCS000002", 
+  "SCPCS000004", 
+  "SCPCS000252", 
+  "SCPCS000258",
+  "SCPCS000262", 
+  "SCPCS000222", 
+  "SCPCS000223", 
+  "SCPCS000224"
+)
+
+# These are the directories to save to output_dir_analysis_data
+# They should contain processed files organized as expected _AND_ with the bulk quant TSV file
+bulk_projects <- c(
+  "SCPCP000001", 
+  "SCPCP000002", 
+  "SCPCP000006", 
+  "SCPCP000009", 
+  "SCPCP000017"
+)
+
+
+# Define input files ------------------------
+
+# Read whitelist so we can check that all projects are present
+expected_projects <- readLines(opts$project_whitelist)
+
+# Define and check input projects
+input_zips <- list.files(
+  opts$portal_projects_dir, 
+  pattern = "^SCPCP\\d{6}_SINGLE-CELL_SINGLE-CELL-EXPERIMENT_\\d+{4}-\\d+{2}-\\d+{2}\\.zip",
```

Comment:
I'm wondering if it makes sense to update this so it just looks for the project id and then single-cell? At some point in the near future we are updating the download structure from the Portal (see https://github.com/AlexsLemonade/scpca-docs/issues/383 or https://github.com/AlexsLemonade/scpca-docs/issues/381), so I wonder if there is a way to make this script robust to those changes while also working with the current portal?

Reply:
Great point! On it. We'll still have to exclude merged, but we do expect that string to stay in the file name ✔️ . Not clear to me from those issues if we'll still have names in caps or switch to lowercase, so will make this code flexible for either.